src/interface/file.rs

```diff
@@ -182,6 +178,51 @@ impl EmulatedFile {
         Ok(bytes_written)
     }
 
+    // Reads entire file into provided C-buffer
+    pub fn readfile_to_new_string(&self, offset: usize) -> std::io::Result<String> {
+
+        let mut buf = String::new();
```

Comment:
If you want, putting this closer to where it's used (inside the Some match arm) will make it clearer what this is/does.
Also, change the name from buf, it's a string not a [u8]. stringbuf would be fine probably.